solutions/src/main/java/ru/sberbank/school/task11/StreamsImpl.java

```diff
@@ -0,0 +1,134 @@
+package ru.sberbank.school.task11;
+
+import lombok.NonNull;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+/**
+ * Облегченная версия класса {@link Stream}
+ */
+public class StreamsImpl<T> {
+    private List<T> myList = new ArrayList<>();
+
+
+    private StreamsImpl(@NonNull T... elements) {
+        myList = Arrays.asList(elements);
+    }
+
+    private StreamsImpl(@NonNull Collection<? extends T> elements) {
+        myList.addAll(elements);
+    }
+
+    /**
+     * Принимает на вход массив элементов и возвращает стрим построенный на основе этих элементов.
+     * @param elements входные элементы
+     * @return стрим элементов из elements
+     */
+    public static <T> StreamsImpl<T> of(@NonNull T... elements) {
+        return new StreamsImpl(elements);
+    }
+
+    /**
+     * Принимает на вход коллекцию элементов и возвращает стрим построенный на основе этих элементов.
+     *
+     * @param elements входные элементы
+     * @return стрим элементов из elements
+     */
+    public static <T> StreamsImpl<T> of(@NonNull Collection<T> elements) {
+        return new StreamsImpl(elements);
+    }
+    /**
+     * Фильтрация элементов по заданному правилу. Необходимо подобрать нужный интерфейс для передачи в этот метод.
+     * Иными словами, этот метод оставляет в коллекции только те элементы, которые удовлетворяют условию в лямбде.
+     * Аналог Stream#map(Function).
+     * Intermediate операция.
+     * @param predicate - правило фильтрации элементов.
+     * @return стрим
+     */
+    public StreamsImpl<T> filter(@NonNull Predicate<? super T> predicate) {
```

Comment:
&#39;METHOD_DEF&#39; should be separated from previous statement.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1369306647>*